pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,376 @@
+                this.kpatchService.enable().then(() => {
+                    this.kpatchService.start().then(() => {
+                        this.setState({ showModal: false, error: "" });
+                    });
+                });
+            })
+                    .catch(e => this.setState({ error: e.toString() }))
+                    .finally(() => {
+                        this.checkSetup();
+                        this.setState({ updating: false });
+                    });
+        } else {
+            cockpit.spawn(["dnf", "-y", "kpatch", "manual"], { superuser: "require", err: "message" }).then(() => {
+                this.kpatchService.disable().then(() => {
+                    this.kpatchService.stop().then(() => {
+                        this.setState({ showModal: false, error: "" });
+                    });
+                });
+            })
+                    .catch(e => this.setState({ error: e.toString() }))
+                    .finally(() => {
+                        this.checkSetup();
+                        this.setState({ updating: false });
+                    });
+        }
+    }
+
+    render() {
+        // Not yet recognized
+        if (this.state.loaded === false || this.state.patchName === null)
+            return null;
+
+        // Not supported on this system
+        if (this.state.unavailable.length > 0)
+            return null;
+
+        let state = _("Enabled");
+        let actionText = _("Edit");
+        let action = () => this.setState({ showModal: true });
+
+        if (this.state.missing.length > 0 && this.state.unavailable.length === 0) {
+            state = _("Not installed");
+            actionText = _("Install");
+            action = this.handleInstall;
+        } else if (!this.state.enabled) {
+            state = _("Disabled");
+            actionText = _("Enable");
+        }
+
+        let kernel_name = "";
+        if (this.state.kernelName)
+            kernel_name = " (" + this.state.kernelName + ")";
+
+        let error = null;
+        if (this.state.error)
+            error = <Alert variant='danger' isInline title={this.state.error} />;
+
+        const body = <Checkbox id="apply-kpatch"
+                               isChecked={this.state.applyCheckbox}
+                               label={_("Apply kernel patches")}
+                               onChange={checked => this.setState({ applyCheckbox: checked })}
+                               body={<>
+                                   <Radio id="current-future" label={_("for current and future kernels")} onChange={() => this.setState({ justCurrent: false })} isDisabled={!this.state.applyCheckbox} isChecked={!this.state.justCurrent} />
+                                   <Radio id="current-only" label={_("for current kernel only") + kernel_name} onChange={() => this.setState({ justCurrent: true })} isDisabled={!this.state.applyCheckbox} isChecked={this.state.justCurrent} />
```

Comment:
Do you mind breaking these long lines and put every property  on a single line? Easier to read.